### PR TITLE
Add tests to StdLibSpec checking comparison between Date & Timestamp …

### PR DESCRIPTION
**Pull request by rabbitonweb**
When comparing `Timestamp` and `Date` returned value should always be `Data.NA`

Closes https://github.com/quasar-analytics/quasar/issues/1778

### CODE REVIEW

**Comment by rabbitonweb** [comment]
This test fails for most of the connectors. 
@sellout @drostron @wemrysi can you check this on your side?

**Comment by sellout** [comment]
See, this is the kind of case where we really want universals (and type classes) in our type system. Rather than having ~ `val LT(a: Comparable, b: Comparable)`, we should have `val LT[A: Comparable](a: A, b: A)`. Which would allow the typechecker to 1. check the constraint and 2. check that the types match. But currently it just checks that each argument independently satisfies the constraint. So, ideally, a connector should never execute a case where the types don’t line up correctly (because the type check would prevent it).

**Comment by rabbitonweb** [comment]
@sellout that would be nice indeed. But for now, we should fix connectors right?

**Comment by sellout** [comment]
Yes, absolutely. I was just tying it back to your new book 😄

**Comment by rabbitonweb** [comment]
@sellout I know :)

**Comment by djspiewak** [comment]
@rabbitonweb Can you assign this PR to someone?

**Comment by rabbitonweb** [comment]
@djspiewak I'm happy to assign but shouldn't I wait till every connector will fix its errors?

**Comment by drostron** [comment]
I have a branch with the potential fix for CB [here](https://github.com/drostron/quasar/tree/std_comparison_CB). Some attention is still due to address a few integration test failures that cropped up due to the change. I've been pulled away by a few other issues but will try and get back to this before too long.

**Comment by alissapajer** [comment]
Can we make these tests pending for the connectors the tests are failing under? Then we could get this merged.

**Comment by rabbitonweb** [comment]
@alissapajer @djspiewak I can make those tests pending, np

File: frontend/src/test/scala/quasar/std/StdLibSpec.scala [hunk 1]
```diff
@@ -1090,7 +1090,9 @@ abstract class StdLibSpec extends Qspec {
           binary(Lte(_, _).embed, Data.Str(x), Data.Str(y), Data.Bool(x <= y))
         }
 
-        // TODO: Timestamp, Interval, cross-type comparison
+        "any date & timestamp" >> prop { (d: LocalDate, i: Instant) =>
+          binary(Lte(_, _).embed, Data.Date(d), Data.Timestamp(i), Data.NA)
```
Comment (alissapajer) [inline frontend/src/test/scala/quasar/std/StdLibSpec.scala]: I think we should also have a test for `Lt` and `Eq`, just to cover all our bases.

**Comment by alissapajer** [comment]
@rabbitonweb I'm working on `StdLibSpec` for mimir now, so since I'm already in this code, I'd be happy to make the failing tests pending so we can get this merged. Then as I mentioned in #1778 we can open individual tickets for each backend. It will be easier to track the remaining work that way, and then we don't have to have PRs open like this for a long time.

File: frontend/src/test/scala/quasar/std/StdLibSpec.scala (hunk 1 above)
Comment (rabbitonweb) [inline frontend/src/test/scala/quasar/std/StdLibSpec.scala]: those are added today

**Comment by rabbitonweb** [comment]
I;ve discussed with @wemrysi possible ways to make it pending for particular connector, will have it done tomorror

**Comment by rabbitonweb** [comment]
@alissapajer ready for review

**Comment by alissapajer** [comment]
@rabbitonweb no idea why but the travis build didn't even run for this one. Will merge when we see a green build. I'll look into it as well.

**Comment by rabbitonweb** [comment]
I'm pretty sure I've seen it green on Travis. But sure, re-run it, np. 
Im afk
I will be back in 30min.

**Comment by alissapajer** [comment]
@rabbitonweb in the most recent build for this PR https://travis-ci.org/quasar-analytics/quasar/builds/253134766 ` some time-related specs are failing in `quasar.std.SimplifyStdLibSpec` for all mongo backends, so they probably just need to be make pending too.

**Comment by alissapajer** [comment]
since @rabbitonweb is off on holiday starting tomorrow, I'm going to take charge of fixing the remaining compile errors so we can get this PR merged. I'll assign another reviewer before merge so I'm not merging my own work :)

File: frontend/src/test/scala/quasar/std/SimplifyStdLibSpec.scala [hunk 2]
```diff
@@ -41,6 +41,12 @@ class SimplifyStdLibSpec extends StdLibSpec {
   def shortCircuit[N <: Nat](func: GenericFunc[N], args: List[Data]): Result \/ Unit = (func, args) match {
     case (relations.Between, _) => notHandled
```
Comment (rabbitonweb) [inline frontend/src/test/scala/quasar/std/SimplifyStdLibSpec.scala]: wow that was in front of my eyes the whole time...

**Comment by alissapajer** [comment]
Taking notes here, newly added tests are still failing in:
`quasar.physical.mongodb.MongoDbJsStdLibSpec`
`quasar.physical.mongodb.MongoDbExprStdLibSpec`
`quasar.physical.sparkcore.fs.CoreMapStdLibSpec`

**Pull request merged**